src/ws/hooks/app.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * Graasp websocket client
+ * React effect hooks to subscribe to real-time updates and mutate query client
+ */
+import { List } from 'immutable';
+import { useEffect } from 'react';
+
+import { UUID, convertJs } from '@graasp/sdk';
+import { AppActionRecord, AppDataRecord, AppSettingRecord } from '@graasp/sdk/frontend';
+
+import { Channel, WebsocketClient } from '../ws-client';
+import { useQueryClient } from '@tanstack/react-query';
+import { buildAppActionsKey, buildAppDataKey, buildAppSettingsKey } from '../../config/keys';
+import { AppActionEvent, AppDataEvent, AppSettingEvent } from '../types';
+import { APP_ACTIONS_TOPIC, APP_DATA_TOPIC, APP_SETTINGS_TOPIC } from '../constants';
+
+export const configureWsAppHooks = (websocketClient: WebsocketClient) => {
+  return {
+    /**
+     * React hook to subscribe to the updates of the app data for
+     * the given item ID.
+     * @param itemId The ID of the item of which to observe updates
+     */
+    useAppDataUpdates: (itemId?: UUID | null) => {
+      const queryClient = useQueryClient();
+      useEffect(() => {
+        if (!itemId || itemId?.length === 0) {
+          return () => {
+            // do nothing
+            console.warn('No correct itemId provided.');
+          };
+        }
+
+        const channel: Channel = { name: itemId, topic: APP_DATA_TOPIC };
+        const appDataKey = buildAppDataKey(itemId);
+
+        const handler = (event: AppDataEvent) => {
+          if (event.kind === 'app-data') {
+            const appDataList: List<AppDataRecord> | undefined =
+              queryClient.getQueryData(appDataKey);
+            const newAppData: AppDataRecord = convertJs(event.appData);
+            switch (event.op) {
+              case 'post': {
+                if (appDataList?.findIndex(({ id }) => id === newAppData.id) === -1)
+                  queryClient.setQueryData(appDataKey, appDataList?.push(newAppData));
```

Comment:
What about the `appDataList` is null or undefined case? Consider creating a new list?